src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/DefaultMappingConfigProvider.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Contrib.Data.Mapping.Mapster;
+
+public class DefaultMappingConfigProvider : IMappingConfigProvider
+{
+    private readonly ConcurrentDictionary<(Type SourceType, Type DestinationType, MapOptions? mapOptions), TypeAdapterConfig?>
+        _store = new();
+
+    private readonly MapOptions _options;
+
+    public DefaultMappingConfigProvider(MapOptions options) => _options = options;
+
+    public TypeAdapterConfig GetConfig(Type sourceType, Type destinationType, MapOptions? options = null)
+        => GetConfigByCache(sourceType, destinationType, options);
+
+    protected virtual TypeAdapterConfig GetConfigByCache(Type sourceType, Type destinationType, MapOptions? options)
+    {
+        TypeAdapterConfig? config = _store.GetOrAdd(
```

Comment:
Why not use lazy or ManualMemoryCache?

Reply:
I don't quite understand what you mean, here is the configuration that tries to get the mapping relationship when the user is mapped